3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/AnonIpEnrichment.scala

```diff
@@ -77,22 +87,61 @@ object AnonOctets extends Enumeration {
 
   /**
    * Convert a Stringly-typed integer
-   * into the corresponding AnonOctets
+   * into the corresponding AnonIPv4Octets
    * Enum Value.
    *
    * Update the Validation Error if the
    * conversion isn't possible.
    *
-   * @param anonOctets A String holding
+   * @param anonIPv4Octets A String holding
    *        the number of IP address
    *        octets to anonymize
-   * @return a Validation-boxed AnonOctets
+   * @return a Validation-boxed AnonIPv4Octets
    */
-  def fromInt(anonOctets: Int): ValidatedMessage[AnonOctets] =
+  def fromInt(anonIPv4Octets: Int): ValidatedMessage[AnonIPv4Octets] =
     try {
-      AnonOctets(anonOctets).success
+      AnonIPv4Octets(anonIPv4Octets).success
     } catch {
-      case nse: NoSuchElementException => "IP address octets to anonymize must be 1, 2, 3 or 4".toProcessingMessage.fail
+      case nse: NoSuchElementException =>
+        "IPv4 address octets to anonymize must be 1, 2, 3 or 4".toProcessingMessage.fail
+    }
+}
+
+/**
+ * How many segments (ipv6) to anonymize?
+ */
+object AnonIPv6Segments extends Enumeration {
+
+  type AnonIPv6Segments = Value
+
+  val One   = Value(1, "1")
+  val Two   = Value(2, "2")
+  val Three = Value(3, "3")
+  val Four  = Value(4, "4")
+  val Five  = Value(5, "5")
+  val Six   = Value(6, "6")
+  val Seven = Value(7, "7")
+  val All   = Value(8, "8")
+
+  /**
+   * Convert a Stringly-typed integer
+   * into the corresponding AnonIPv6Segments
+   * Enum Value.
+   *
+   * Update the Validation Error if the
+   * conversion isn't possible.
+   *
+   * @param anonSegments A String holding
+   *        the number of IPv6 address
+   *        segments to anonymize
+   * @return a Validation-boxed AnonIPv6Segments
+   */
+  def fromInt(anonIPv6Segments: Int): ValidatedMessage[AnonIPv6Segments] =
+    try {
```

Comment:
I'd stay away from `try`

Reply:
Totally agree, but similar code is in all enrichments, so I wanted to keep it consistent.